hub/core/storage/lru_cache.py

```diff
@@ -105,7 +114,25 @@ def get_hub_object(
         Returns:
             An instance of `expected_class` populated with the data.
         """
-
+        if get_partial:
+            assert issubclass(expected_class, BaseChunk)
+            if path in self.lru_sizes:
+                return self[path]
+            last_end_byte = 10 * KB
+            item = self.get_bytes(path, 0, last_end_byte)
+            while 1:
```

Comment:
how many times this loops over a single request typically?

Reply:
it should only happen once for these chunks which have between 2 and 20 samples (for which we're doing partial requests) but this is just for future cases if we extend partial reads to smaller samples

Reply:
@khustup has some good heuristics in visualizer how to fetch metadata first, then minimize number of requests the fetch the data. This becomes especially important for labels or small data where instead of single byte request you would do optimistic request.

Reply:
we can do such an approach as well, but seems like overkill when this is a temporary solution right now.
For now, we fetch full chunks for smaller objects and partial reads for bigger ones.

Reply:
sounds good